### PR TITLE
Revamp EVM context flow, hash fixes, cleanups

### PR DESCRIPTION
**Pull request by prasannavl**
## Summary

- Uses a single context carry for EVM enable check to simplify things. 
- EVM enables and disables doesn't affect the current block with the `setgov`. Takes effect fully in the next block. 
- Introduces `FromByteArray` to avoid confusion from and to `base_blob` which internally stores bytes in the reversed order. 
- Add more cross boundary helpers.
- Fixes many things, including reversed hashes, miner mining invalid blocks due to missing result checks, etc  
- Few misc cleanups

TODO: 
- Fix migration and block hash change related to it

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [x] Includes consensus refactors
  - [ ] None

### CODE REVIEW

File: src/masternodes/validation.cpp [hunk 1]
```diff
@@ -2381,9 +2381,12 @@ static void RevertFailedTransferDomainTxs(const std::vector<std::string> &failed
 }
 
 static Res ValidateCoinbaseXVMOutput(const XVM &xvm, const FinalizeBlockCompletion &blockResult) {
-    const auto coinbaseBlockHash = uint256(std::vector<uint8_t>(blockResult.block_hash.begin(), blockResult.block_hash.end()));
+    const auto blockResultBlockHash = uint256::FromByteArray(blockResult.block_hash);
+    
+    // LogPrintf("DEBUG:: EVMQ Validate: blockResult: %s\n", blockResultBlockHash.ToString());
+    // LogPrintf("DEBUG:: EVMQ Validate: xvm: %s\n", xvm.ToUniValue().write());
```
Comment (Jouzo) [inline src/masternodes/validation.cpp:2387]:
```suggestion
```

File: src/miner.cpp [hunk 2]
```diff
@@ -389,18 +391,16 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
...
-
-            CDataStream metadata(SER_NETWORK, PROTOCOL_VERSION);
-            metadata << xvm;
-
-            CScript script;
-            script << OP_RETURN << ToByteVector(metadata);
-
-            coinbaseTx.vout[headerIndex].scriptPubKey = script;
+            coinbaseTx.vout[headerIndex].scriptPubKey = xvm.ToScript();
+            // LogPrintf("DEBUG:: CreateNewBlock:: xvm-coinbase-add:: %s\n", XVM::TryFrom(xvm.ToScript())->ToUniValue().write());
```
Comment (Jouzo) [inline src/miner.cpp:403]:
```suggestion
```

File: src/miner.cpp [hunk 3]
```diff
@@ -39,21 +39,21 @@
 #include <random>
 #include <utility>
 
-struct EvmAddressWithNonce {
-    EvmAddress address;
+struct EvmAddressDataWithNonce {
```
Comment (Jouzo) [inline src/miner.cpp:42]: Would keep the names rather than types, i.e. `EvmAddressWithNonce`

**Review by Jouzo — APPROVED** [review]

File: src/miner.cpp (hunk 3 above)
Comment (prasannavl) [reply src/miner.cpp:42]: Good catch, agree, will change

**Review by sieniven — DISMISSED** [review]

**Pull request merged**